binder/src/main/scala/PanamaCIRCT.scala

```diff
@@ -0,0 +1,394 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.internal.panama.circt
+
+import java.lang.foreign._
+import java.lang.foreign.MemorySegment.NULL
+import java.lang.foreign.ValueLayout._
+
+import org.llvm.circt
+import org.llvm.circt.CAPI
+
+// Wrapper for CIRCT APIs with Panama framework
+class PanamaCIRCT {
+  // Open an arena for memory management of MLIR API calling in this context instance
+  private val arena = Arena.openConfined()
+
+  // Create MLIR context and register dialects we need
+  private val mlirCtx = {
```

Comment:
Think about it:
for each ctx, how to record current Dialect, not firrtl, chirrtl only, but `chirrtl::CHIRRTLDialect, firrtl::FIRRTLDialect, hw::HWDialect, comb::CombDialect, seq::SeqDialect, om::OMDialect, sv::SVDialect, verif::VerifDialect, ltl::LTLDialect`,
how to record current transform(currently unimplemented)?
Is possible to `copy` an ctx(may not be possible), so we can pass `firrtl` to `verilog` and `arc` dialect at same time.